File: packages/observability/examples/python-ingest/push.py
"""Push 模式示例：向 Toonflow ingest 上报日志"""
import os
import json
import urllib.request

INGEST = os.environ.get("TOONFLOW_LOG_INGEST", "http://localhost:10588/api/logs/ingest")
TOKEN = os.environ.get("TOONFLOW_TOKEN", "")

event = {
    "level": "error",
    "category": "ai_call",
    "message": "simulated vendor failure",
    "vendorId": "agnesai",
    "traceId": "demo-trace-python",
    "payload": {"errorCategory": "rate_limit"},
}

req = urllib.request.Request(
    INGEST,
    data=json.dumps(event).encode("utf-8"),
    headers={"Content-Type": "application/json", "Authorization": f"Bearer {TOKEN}"},
    method="POST",
)
with urllib.request.urlopen(req) as resp:
    print(resp.status, resp.read().decode())

File: packages/observability/examples/push-node/push.ts
import { createPushClient } from "@toonflow/observability";

async function main() {
  const push = createPushClient({
    endpoint: process.env.TOONFLOW_LOG_INGEST || "http://localhost:10588/api/logs/ingest",
    getToken: () => process.env.TOONFLOW_TOKEN || "",
  });

  await push.ingest({
    level: "error",
    category: "system",
    message: "demo from node push client",
    traceId: "demo-node-push",
  });

  console.log("pushed");
}

main().catch((e) => {
  console.error(e);
  process.exit(1);
});
